discretisedfield/field.py

```diff
@@ -3479,3 +3480,210 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        >>> xa.sel(comp='x')
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+            comp     <U1 'x'
+        Attributes:
+            units:    None
+
+        """
+        if not isinstance(name, str):
+            msg = "Name argument must be a string."
+            raise TypeError(msg)
+
+        if units is not None and not isinstance(units, str):
+            msg = "Units argument must be a string."
+            raise TypeError(msg)
+
+        axes = ['x', 'y', 'z']
+
+        data_array_coords = {
+            axis: np.fromiter(self.mesh.axis_points(axis), dtype=float)
+            for axis in axes
+        }
+
+        if 'unit' in self.mesh.attributes:
+            geo_units_dict = dict.fromkeys(axes, self.mesh.attributes['unit'])
+        else:
+            geo_units_dict = dict.fromkeys(axes, 'm')
+
+        if self.dim > 1:
+            data_array_dims = axes + ['comp']
+            if self.components is not None:
+                data_array_coords['comp'] = self.components
+            field_array = self.array
+        else:
+            data_array_dims = axes
+            field_array = np.squeeze(self.array, axis=-1)
+
+        data_array = xr.DataArray(field_array,
+                                  dims=data_array_dims,
+                                  coords=data_array_coords,
+                                  name=name,
+                                  attrs=dict(units=units))
+
+        for dim in geo_units_dict:
+            data_array[dim].attrs['units'] = geo_units_dict[dim]
+
+        return data_array
+
+    @classmethod
+    def from_xarray(cls, xa):
+        if not isinstance(xa, xr.DataArray):
+            msg = "Argument must be a xr.DataArray."
+            raise TypeError(msg)
+
+        if xa.ndim not in [3, 4]:
+            msg = "DataArray dimensions must be 3 or 4."
+            raise ValueError(msg)
+
+        if xa.ndim == 3:
+            for dim in xa.dims:
+                if dim not in ['x', 'y', 'z']:
+                    msg = "The dimensions must be either 'x', 'y', or 'z'."
+                    raise ValueError(msg)
+        elif xa.ndim == 4:
+            for dim in xa.dims:
+                if dim not in ['x', 'y', 'z', 'comp']:
+                    msg = "The dimensions must be either "\
+                          "'x', 'y', 'z' or 'comp'."
+                    raise ValueError(msg)
+            if xa['comp'].values.size != len(set(xa['comp'].values)):
+                msg = "All the components should be unique."
+                raise ValueError(msg)
+            if 'comp' in xa.coords:
+                for comp in xa['comp'].values:
+                    if not isinstance(comp, str):
```

Comment:
`if 'comp' in xa.coords and any(not isinstance(comp, str) for comp in xa['comp'].values)`

Reply:
Good point! But I will not be able to use `elif` here which checks, even if `comp` are `str`, they should be one of `xyz`?